### PR TITLE
Update app framework to serve GraphQL api doc

### PR DESCRIPTION
**Pull request by byliuyang**
### Description
Previously, external developers have to read code in order to understand how to consume Short APIs. Now, the will be able to see API doc at `gql.short-d.com`.

### Screenshots
Screen Shot 2020-06-27 at 1 03 20 AM

### CODE REVIEW

File: backend/dep/provider/graphql.go [hunk 1]
```diff
@@ -11,6 +11,17 @@ type GraphQLPath string
...
+	webUI graphql.WebUI,
+	logger logger.Logger,
+) service.GraphQL {
+	return service.NewGraphQL(logger, string(gqlPath), handler, webUI)
+}
+
+type GraphiQLDefaultQuery string
+
+func NewGraphiQL(gqlPath GraphQLPath, defaultQuery GraphiQLDefaultQuery) graphql.GraphiQL {
+	return graphql.NewGraphiQL(string(gqlPath), string(defaultQuery))
```
Comment (byliuyang) [inline backend/dep/provider/graphql.go:26]: Need to add comments here.

File: backend/app/app.go [hunk 2]
```diff
@@ -36,6 +36,7 @@ type ServiceConfig struct {
 	SearchTimeout        time.Duration
 	SwaggerUIDir         string
 	OpenAPISpecPath      string
+	GraphiQLDefaultQuery string
```
Comment (arberiii) [inline backend/app/app.go:39]: Why it is named Graph**i**QL, shouldn't it be without "i"?

File: backend/main.go [hunk 3]
```diff
@@ -44,6 +44,7 @@ func main() {
 		SearchTimeout        time.Duration `env:"SEARCH_TIMEOUT" default:"1s"`
 		SwaggerUIDir         string        `env:"SWAGGER_UI_DIR" default:"app/adapter/routing/public"`
 		OpenAPISpecPath      string        `env:"OPEN_API_SPEC_PATH" default:"app/adapter/routing/api.yml"`
+		GraphiQLDefaultQuery string        `env:"GRAPH_I_QL_DEFAULT_QUERY" default:""`
```
Comment (arberiii) [inline backend/main.go:47]: Please add it also in env.dist file

File: backend/dep/provider/graphql.go [hunk 4]
```diff
@@ -11,6 +11,17 @@ type GraphQLPath string
 
 // NewGraphQLService creates GraphQL service with GraphQlPath to uniquely
 // identify graphqlPath during dependency injection.
```
Comment (arberiii) [inline backend/dep/provider/graphql.go:13]:
```suggestion
// NewGraphQLService creates GraphQL service with GraphQLPath to uniquely
// identify graphQLPath during dependency injection.
```

File: backend/dep/provider/graphql.go [hunk 5]
```diff
@@ -11,6 +11,17 @@ type GraphQLPath string
...
+func NewGraphQLService(
+	gqlPath GraphQLPath,
+	handler graphql.Handler,
+	webUI graphql.WebUI,
+	logger logger.Logger,
+) service.GraphQL {
+	return service.NewGraphQL(logger, string(gqlPath), handler, webUI)
+}
+
+type GraphiQLDefaultQuery string
```
Comment (arberiii) [inline backend/dep/provider/graphql.go:23]: Comment here!

**Comment by byliuyang** [comment]
@arberiii It's built into app framework.

**Pull request merged**